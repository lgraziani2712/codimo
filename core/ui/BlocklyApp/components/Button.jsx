/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import { COLOR_PALETTE } from 'core/constants/colors';

const AnimatedButton = styled.a`
  border-radius: 10px;
  cursor: ${({ disabled }: Props) => (disabled ? 'cursor' : 'pointer')};
  float: left;
  font-size: 25px;
  font-weight: bold;
  opacity: ${({ disabled }: Props) => (disabled ? '0.6' : 'auto')};
  padding: 10px 40px;
  pointer-events: ${({ disabled }: Props) => (disabled ? 'none' : 'auto')};
  text-decoration: none;
  transition: all 0.1s;
  &:active {
    border-bottom: 1px solid;
    transform: translate(0, 5px);
  }
`;
const ColoredAnimatedButton = styled(AnimatedButton)`
  background-color: ${({ type }: Props) => COLOR_PALETTE[type].normal};
  border-bottom: 5px solid ${({ type }: Props) => COLOR_PALETTE[type].dark};
  color: ${({ type }: Props) => COLOR_PALETTE[type].color};
  text-shadow: 0 -2px ${({ type }: Props) => COLOR_PALETTE[type].dark};
`;

type Props = {|
  disabled?: boolean,
  title: string,
  type: $Keys<typeof COLOR_PALETTE>,
  handleClick: (event: SyntheticEvent) => void,
|};
export default function Button({ handleClick, title, type, disabled }: Props) {
  return (
    <ColoredAnimatedButton
      disabled={disabled}
      type={type}
      onClick={handleClick}
    >{title}</ColoredAnimatedButton>
  );
}
