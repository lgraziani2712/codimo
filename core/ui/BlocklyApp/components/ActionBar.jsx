/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import blocklyTextsUI from 'core/constants/localize/es/blocklyTextsUI';

import Button from './Button';

const Bar = styled.div`
  align-items: center;
  background-color: #e3e3e3;
  border: 1px solid #ddd;
  border-bottom: 0;
  display: flex;
  height: 68px;
  width: calc(100% - 2px);

  & * {
    margin: 0 0 0 10px;
  }
`;

type Props = {|
  isExecuting: boolean,
  isStopped: boolean,
  handleStartGame(): void,
  handleResetGame(): void,
|};
const ActionBar = ({ isExecuting, isStopped, handleStartGame, handleResetGame }: Props) => (
  <Bar>
    {isStopped ?
      <Button
        title={blocklyTextsUI.actions.play}
        type="green"
        handleClick={handleStartGame}
      /> :
      <Button
        disabled={isExecuting}
        title={blocklyTextsUI.actions.reset}
        type="orange"
        handleClick={handleResetGame}
      />
    }
  </Bar>
);

export default ActionBar;
