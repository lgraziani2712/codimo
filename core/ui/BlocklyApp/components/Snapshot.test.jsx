/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

import ActionBar from './ActionBar';
import Button from './Button';

describe('ui/BlocklyApp/components', () => {
  it('renders the Button correctly', () => {
    const buttonTree = renderer.create(
      <Button handleClick={() => {}} title="I'm a button" type="red" />,
    ).toJSON();

    expect(buttonTree).toMatchSnapshot();
  });
  it('renders the ActionBar correctly', () => {
    const tree = renderer.create(
      <ActionBar
        isExecuting={false}
        isStopped={true}
        handleStartGame={() => {}}
        handleResetGame={() => {}}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
