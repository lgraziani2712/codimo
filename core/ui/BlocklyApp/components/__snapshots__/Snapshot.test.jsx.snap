// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/BlocklyApp/components renders the ActionBar correctly 1`] = `
.c2 {
  border-radius: 10px;
  cursor: pointer;
  float: left;
  font-size: 25px;
  font-weight: bold;
  opacity: auto;
  padding: 10px 40px;
  pointer-events: auto;
  text-decoration: none;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
}

.c2:active {
  border-bottom: 1px solid;
  -webkit-transform: translate(0,5px);
  -ms-transform: translate(0,5px);
  transform: translate(0,5px);
}

.c1 {
  background-color: #5cb85c;
  border-bottom: 5px solid #1b7c1b;
  color: #fff;
  text-shadow: 0 -2px #1b7c1b;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #e3e3e3;
  border: 1px solid #ddd;
  border-bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 68px;
  width: calc(100% - 2px);
}

.c0 * {
  margin: 0 0 0 10px;
}

<div
  className="c0"
>
  .c1 {
  border-radius: 10px;
  cursor: pointer;
  float: left;
  font-size: 25px;
  font-weight: bold;
  opacity: auto;
  padding: 10px 40px;
  pointer-events: auto;
  text-decoration: none;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
}

.c1:active {
  border-bottom: 1px solid;
  -webkit-transform: translate(0,5px);
  -ms-transform: translate(0,5px);
  transform: translate(0,5px);
}

.c0 {
  background-color: #5cb85c;
  border-bottom: 5px solid #1b7c1b;
  color: #fff;
  text-shadow: 0 -2px #1b7c1b;
}

<a
    className="c0 c1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<img class=\\"emoji\\" draggable=\\"false\\" alt=\\"â–¶\\" src=\\"https://twemoji.maxcdn.com/2/svg/25b6.svg\\"/> PROBAR",
      }
    }
    disabled={undefined}
    onClick={[Function]}
    type="green"
  />
</div>
`;

exports[`ui/BlocklyApp/components renders the Button correctly 1`] = `
.c1 {
  border-radius: 10px;
  cursor: pointer;
  float: left;
  font-size: 25px;
  font-weight: bold;
  opacity: auto;
  padding: 10px 40px;
  pointer-events: auto;
  text-decoration: none;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
}

.c1:active {
  border-bottom: 1px solid;
  -webkit-transform: translate(0,5px);
  -ms-transform: translate(0,5px);
  transform: translate(0,5px);
}

.c0 {
  background-color: #d9534f;
  border-bottom: 5px solid #be2f2b;
  color: #fff;
  text-shadow: 0 -2px #be2f2b;
}

<a
  className="c0 c1"
  dangerouslySetInnerHTML={
    Object {
      "__html": "I'm a button",
    }
  }
  disabled={undefined}
  onClick={[Function]}
  type="red"
/>
`;
